### ISSUE TITLE
Nullable parameter is incorrectly parsed in a lambda

### ISSUE THREAD

**Issue by CorentinDeBoisset** [astral-sh/ty#792]
### Summary

```py
from typing import Iterable

def sort(iterable: Iterable, attribute: str | None = None) -> Iterable:
    if not attribute:
        return sorted(iterable)
    else:
        return sorted(iterable, key=lambda item: item.get(attribute) if type(item) is dict else getattr(item, attribute))
```

This error is raised on the final line:

```
Argument to function `getattr` is incorrect: Expected `str`, found `str | None` (invalid-argument-type)
```

Playground: https://play.ty.dev/caa55b6f-6506-405b-acaf-a71fdbe7d3a7

### Version

ty 0.0.1-alpha.13 (6a408d5f9 2025-07-02)

**Comment by AlexWaygood** [comment]
A smaller repro is:

```py
def f(x: str): ...

def g(x: str | None):
    if not x:
        return

    def h():
        f(x)

    h()  # Argument to function `f` is incorrect: Expected `str`, found `str | None` (invalid-argument-type)
```

I think we already have an issue open for this somewhere

**Comment by AlexWaygood** [comment]
Yeah, I think this is a duplicate of #559. Thanks for the report, though!

**Comment by CorentinDeBoisset** [comment]
Actually, after tinkering with it a bit, it seems that python lambda are absolutely not proper closures. See [this playground](https://play.ty.dev/53cc3faf-781c-4aeb-b189-7494c1528dd9). No idea what should be the proper course of action.
